src/test/protocol/STValidation_test.cpp

```diff
@@ -63,32 +73,28 @@ class STValidation_test : public beast::unit_test::suite
             0x00, 0x8a, 0x00, 0x88, 0x00, 0xe6
         };
 
-        try
         {
-            SerialIter sit{payload1, sizeof(payload1)};
+            SerialIter sit{payload0, sizeof(payload0)};
             auto stx = std::make_shared<ripple::STValidation>(sit,
                 [](PublicKey const& pk) {
                     return calcNodeID(pk);
                 }, false);
-            fail("An exception should have been thrown");
         }
-        catch (std::exception const& e)
+
         {
-            BEAST_EXPECT(strcmp(e.what(), "Invalid public key in validation") == 0);
+            SerialIter sit{payload1, sizeof(payload1)};
+            auto stx = std::make_shared<ripple::STValidation>(sit,
+                [](PublicKey const& pk) {
+                    return calcNodeID(pk);
+                }, false);
         }
 
-        try
         {
             SerialIter sit{payload2, sizeof(payload2)};
             auto stx = std::make_shared<ripple::STValidation>(sit,
                 [](PublicKey const& pk) {
                     return calcNodeID(pk);
                 }, false);
-            fail("An exception should have been thrown");
```

Comment:
payload3 causes the exception

Reply:
It might be worth adding a public key with a bad signature and checking that exception...
```
        // Invalid signature
        constexpr unsigned char payload4[] = {
            0x73, 0x21, 0xed, 0xff, 0x03, 0x1c, 0xbe, 0x65, 0x22,
             0x61, 0x9c, 0x5e, 0x13, 0x12, 0x00, 0x3b, 0x43, 0x00,
             0x00, 0x00, 0xf7, 0x3f, 0x3f, 0x3f, 0x3f, 0x3f, 0x3f,
             0x3f, 0x3f, 0x13, 0x13, 0x13, 0xaa, 0xaa, 0xaa};
        try
        {
            SerialIter sit{payload4, sizeof(payload4)};
            std::make_shared<ripple::STValidation>(
                sit, [](PublicKey const& pk) { return calcNodeID(pk); }, true);
            fail("An exception should have been thrown");
        }
        catch (std::exception const& e)
        {
            BEAST_EXPECT(
                strcmp(e.what(), "Invalid signature in validation") == 0);
        }
```